### PR TITLE
[AE-26] Increase Library name length from 16 to 32

### PR DESCRIPTION
**Pull request by aliphys**
As reported in https://github.com/arduino/arduino-lint/issues/508 , the current linter rules limit the max number of characters for a library name to 16. 

https://github.com/arduino/arduino-lint/blob/c2c82d654e1cf2888112cfc83e3417342b163227/etc/schemas/arduino-library-properties-definitions-schema.json#L94

https://github.com/arduino/arduino-lint/blob/c2c82d654e1cf2888112cfc83e3417342b163227/internal/rule/ruleconfiguration/ruleconfiguration.go#L427

As noted by @aentinger , since Arduino libraries start with `Arduino_`, this leaves only 8 characters to describe the library according to the linter rules. **This PR increases the maximum length of a library name to ~64~ 32 characters**. For official libraries with the leading `Arduino_`, this provides ~56~ 24 characters which should be sufficent to describe the library in a UX-friendly manner.

### CODE REVIEW

**Comment by aliphys** [comment]
@per1234 , I saw you self assigned on https://github.com/arduino/arduino-lint/issues/508 .
I increased the character length to 64 in this PR. Could you review it? Thanks :)

**Comment by aentinger** [comment]
Honestly, I feel 64 is a bit _too_ much :wink: .

After all we do not want to encourage `ThisIsMyFancyArduinoLibraryWhichHasATotalLongAndIncromprehensib`(le Name).

I believe 32 would be enough by far.

Secondly, we could add an "exception" for official Arduino libraries, insofar that the "Arduino_" prefix is not counting towards the limit of 16 characters.

@per1234 @aliphys what is your feedback to this?

**Comment by aliphys** [comment]
You bring a very compelling example @aentinger . Changed it to 32 characters. 👍

File: etc/schemas/arduino-library-properties-definitions-schema.json [hunk 1]
```diff
@@ -91,7 +91,7 @@
                 "$ref": "#/definitions/propertiesObjects/name/specification/object"
               },
               {
-                "maxLength": 16
+                "maxLength": 32
```
Comment (per1234) [inline etc/schemas/arduino-library-properties-definitions-schema.json:94]:
```suggestion
                "maxLength": 36
```
I apologize for being slow to review. The Tooling Team discussed this internally and decided that it should be 36.

File: internal/rule/ruleconfiguration/ruleconfiguration.go [hunk 2]
```diff
@@ -424,7 +424,7 @@ var configurations = []Type{
 		ID:               "LP010",
 		Brief:            "name > recommended length",
 		Description:      "The `name` field in the library's `library.properties` metadata file is longer than the recommended length. As the unique identifier for the library, the name will be typed by the users of command line tools (e.g., `arduino-cli lib install Servo`). For this reason, it is best practices to avoid unnecessary name length.",
-		MessageTemplate:  "library.properties name value {{.}} is longer than the recommended length of 16 characters.",
+		MessageTemplate:  "library.properties name value {{.}} is longer than the recommended length of 32 characters.",
```
Comment (per1234) [inline internal/rule/ruleconfiguration/ruleconfiguration.go:427]:
```suggestion
		MessageTemplate:  "library.properties name value {{.}} is longer than the recommended length of 36 characters.",
```

**Review by per1234 — CHANGES_REQUESTED** [review]
@aliphys please update the value here as well:

https://github.com/arduino/arduino-lint/blob/629e77468ceb3b06a21536bb239544773ad8f072/internal/rule/schema/schemadata/bindata.go#L1512

---

Please update the test data here:

https://github.com/arduino/arduino-lint/blob/629e77468ceb3b06a21536bb239544773ad8f072/internal/rule/rulefunction/testdata/libraries/NameGTRecommendedLength/library.properties#L1

File: etc/schemas/arduino-library-properties-definitions-schema.json (hunk 1 above)
Comment (cmaglie) [reply etc/schemas/arduino-library-properties-definitions-schema.json:94]: > Can you share the rationale for "36"?

The suggestion comes from @91volt supported by a visive test on the IDE:

image

I'm fine with `32` too, no strong opinion on this.

**Comment by 91volt** [comment]
32 is fine 👍

**Comment by cmaglie** [comment]
> Secondly, we could add an "exception" for official Arduino libraries, insofar that the "Arduino_" prefix is not counting towards the limit of 16 characters.

I agree, but that should be valid also for other vendors, like `Adafruit_`, `SparkFun_`, `????_`... which have a lot of libraries.

Anyway, I guess that just increasing the limit to `32` should already alleviate it.

**Comment by aentinger** [comment]
Don't get me wrong, "36" is fine too. I was just baffled to be presented with this number out of the blue. If 36 makes sense because its the max amount of chars that can be sensibly displayed by the IDE then this is a reasonable criteria imho.

>I agree, but that should be valid also for other vendors, like Adafruit_, SparkFun_, ????_... which have a lot of libraries.

Fair enough :grin: . It was just a thought. Given that it is "just" a warning leaving it at 16 is also fine, but I think you all know about the dangers of ignoring warnings and [normalization of deviance](https://danluu.com/wat/). Both 36 or 32 would alleviate that pain point for all involved. I'd be fine with going ahead with 36.

**Comment by cmaglie** [comment]
LOL, ok, let's not get this longer than needed...

I decide: let's go with `32` :-)

**Comment by aliphys** [comment]
Since we decided together on `32` characters as the max library length 🎉 , I changed `NameGTRecommendedLength` (23 characters) to `NameIsBiggerThanRecommendedLength` (33 characters).

**Comment by aliphys** [comment]
For the record, Codecov is really useful for the CI/CD testing pipeline. When I changed the schema file (https://github.com/arduino/arduino-lint/pull/511/commits/b0f7c8a27b3eca5a87732cc8e20ccf436028d316) to 32 it reported a loss in code coverage. 
image
This was then resolved immediately after https://github.com/arduino/arduino-lint/pull/511/commits/b0f7c8a27b3eca5a87732cc8e20ccf436028d316 
image

😄

**Review by per1234 — APPROVED** [review]

**Pull request merged**